contracts/TestCaller.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+struct CalledData {
+    uint256 size;
+    address addr;
+    address tokenIdentifier;
+}
+
+contract TestCaller {
+    CalledData[] private calledDataParams;
+
+    constructor() {}
+
+    function upgrade() public {}
+
+    function callPayable() public payable {}
+
+    function callNonPayable() public {}
+
+    function callPayableWithParams(uint256 size, address addr, address tokenIdentifier) public payable {
```

Comment:
I believe we can remove `payable` modifier since we're not transferring Ether. Instead, on Ethereum, we're transferring erc20 token balances between accounts

Reply:
thanks, good observation

Reply:
good observation, indeed :+1: